File: jar-auth/src/main/java/com/beancounter/auth/client/OAuth2Response.java
package com.beancounter.auth.client;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@JsonIgnoreProperties(ignoreUnknown = true)
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class OAuth2Response {
  @JsonProperty("access_token")
  String token;
  @JsonProperty("expires_in")
  long expiry;
  @JsonProperty("refresh_token")
  String refreshToken;
  @JsonProperty("token_type")
  String type;
  String scope;

}

File: jar-auth/src/main/java/com/beancounter/auth/client/OAuth2Response.kt
package com.beancounter.auth.client

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
/**
 * Standard OAuth 2 response to a login request
 */
class OAuth2Response(
    @JsonProperty("access_token") var token: String?,
    @JsonProperty("expires_in") var expiry: Long,
    @JsonProperty("refresh_token") var refreshToken: String?,
    @JsonProperty("token_type") var type: String?,
)
